pyplot/baselineFits/iterativeGausFit.py

```diff
@@ -0,0 +1,284 @@
+
+def getHistoKeys(inFile,cType="", attr1="", attr2=""):
+    inFile.cd()
+    histo_keys=[]
+    for key in inFile.GetListOfKeys():
+        kname = key.GetName()
+        classType = key.GetClassName()
+        if classType.find(cType) != -1 and kname.find(attr1) != -1 and kname.find(attr2) != -1:
+            histo_keys.append(kname)
+    return histo_keys
+              
+def readhistoFromFile(inFile, histo_key):
+    histo = deepcopy(getattr(inFile, histo_key))
+    return histo
+
+def getChannelFitsPlots(inFile,channels):
+    inFile.cd()
+
+def getTupleFlags(inFile,key):
+    inFile.cd()
+    minbinFail=[]
+    noisy=[]
+    lowdaq=[]
+    channels=[]
+    myTree = inFile.gaus_fit
+    for fitData in myTree:
+        SvtAna2DHisto_key = str(fitData.SvtAna2DHisto_key)
+        if key == SvtAna2DHisto_key+"_hh":
+            channels.append(fitData.channel)
+            minbinFail.append(fitData.minbinFail)
+            noisy.append(fitData.noisy)
+            lowdaq.append(fitData.lowdaq)
+    return channels, minbinFail, noisy, lowdaq
+
+
+
+def getGausFitParameters(inFile, key):
+    inFile.cd()
+    histo_key=[]
+    channel=[]
+    mean=[]
+    sigma=[]
+    norm = []
+    range_lower=[]
+    range_upper=[]
+    Chi2=[]
+    Ndf=[]
+
+    myTree = inFile.gaus_fit
+    for fitData in myTree:
+        SvtAna2DHisto_key = str(fitData.SvtAna2DHisto_key)
+        if key == SvtAna2DHisto_key+"_hh":
+            histo_key.append(SvtAna2DHisto_key)
+            channel.append(fitData.channel)
+            mean.append(fitData.BlFitMean)
+            sigma.append(fitData.BlFitSigma)
+            norm.append(fitData.BlFitNorm)
+            range_lower.append(fitData.BlFitRangeLower)
+            range_upper.append(fitData.BlFitRangeUpper)
+            Chi2.append(fitData.BlFitChi2)
+            Ndf.append(fitData.BlFitNdf)
+
+    return channel,mean, sigma, histo_key,norm, range_lower, range_upper, Chi2, Ndf
+               
+def buildTGraph(name,title, n_points, x, y,color):
+    g = r.TGraph(n_points,np.array(x, dtype = float), np.array(y, dtype = float))
+    g.SetName("%s"%(name))
+    g.SetTitle("%s"%(title))
+    g.SetLineColor(color)
+    return g
+
+def getVarWithKey(var,var_length, key, tuple_key, datatype):
+    x=[]
+    for i in range(var_length):
+        if key == tuple_key[i]:
+            x.append(var[i])
+    x=np.array(x,dtype = datatype)
+    if len(x) == 0:
+        print "No tuple_key was found matching input key"
+        return 0
+    return x
+
+def newCanvas(name):
+        canvas = r.TCanvas("%s"%(name), "c", 1800,800)
+        return canvas
+
+def savePNG(canvas,directory,name):
+        canvas.Draw()
+        canvas.SaveAs("%s/%s.png"%(directory,name))
+        canvas.Close()
+
+
+
+######################################################################################################
+directory = "/home/alic/src/hpstr/pyplot/baselineFits/fit_data/"
```

Comment:
Try to avoid hardcoding stuff like this. You should be able to pass this in via command line.